### PR TITLE
Make CI pass

### PR DESCRIPTION
**Pull request by bquorning**
On CI with Ruby 2.6, when running `TestReporterPrivateStartStop#test_symbols_report` or `TestReporterPublicStartStop#test_symbols_report`, the string "this is a string" is retained. This doesn't happen when running `TestReporter#test_symbols_report`.

It’s really weird and I don’t understand it. But I think a weird-looking edge case in the tests is better than constantly failing CI runs.

### CODE REVIEW

**Comment by SamSaffron** [comment]
oh my... this passes 2.6 just to surface an issue with 2.7 :cry:  

Very much appreciate you digging into this.

**Comment by bquorning** [comment]
Wow… this change did pass CI on my fork just 10 minutes ago: https://github.com/bquorning/memory_profiler/actions/runs/1281570985. There’s definitely something weird going on.

**Comment by bquorning** [comment]
Ok, so https://github.com/SamSaffron/memory_profiler/actions/runs/1281582968 failed on Ruby 2.7:
```
  1) Failure:
TestReporterPrivateStartStop#test_counts [/home/runner/work/memory_profiler/memory_profiler/test/test_reporter.rb:81]:
Expected: 1
  Actual: 4
```
I added the seed to TESTOPTS and re-ran the tests. Now [Ruby 2.7 is passing and Ruby 2.6 is failing](https://github.com/SamSaffron/memory_profiler/actions/runs/1281684628) with
```
  1) Failure:
TestReporterPrivateStartStop#test_counts [/home/runner/work/memory_profiler/memory_profiler/test/test_reporter.rb:81]:
Expected: 1
  Actual: 4
```
@SamSaffron do you have any idea about what the problem might be here?

**Comment by SamSaffron** [comment]
This is such a tricky issue cause the GC has determinism issues and they are different for different versions of ruby. 

I wonder if we should just change it so in test we run the full GC even more times to ensure stuff is collected?

File: test/test_reporter.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ def default_block
 
   # Shared method that creates a Results with 1 retained object using options provided
   def create_report(options = {}, &profiled_block)
+    GC.start
```
Comment (SamSaffron) [inline test/test_reporter.rb:29]: note you may need to do this 3 times or even more to fully collect stuff.

Comment (bquorning) [reply test/test_reporter.rb:29]: Yep. I am still unsure where even to put it. I am very confused by the fact that only `test_counts` seems to fail.

Comment (bquorning) [reply test/test_reporter.rb:30]: Ahh, it’s the only test case running `create_report` with its `default_block` *and* making assertions about total_retained.

I am leaning towards just skipping that assertion for Ruby 2.6 😬

**Comment by bquorning** [comment]
I am sorry to make the tests _less_ specific, but this is the best I could come up with.

**Comment by benoittgt** [comment]
Do we still want to have a test for 2.7. Don't we want instead a CI test for 3.1 ?
